ipi/engine/motion/dynamics.py

```diff
@@ -283,6 +283,10 @@ def bind(self, motion):
         self._dt = motion._dt
         self._nmts = motion._nmts
 
+        # calculate active atoms
+        full_indices = np.arange(len(self.beads.p[0]))
+        self.activeatoms_dof = np.setdiff1d(full_indices, self.fixatoms_dof)
+
```

Comment:
I think this place to get the active dof is ok for the reasons you described.
Indeed, I would suggest adding a check here to verify if the indices are compatible with the number of atoms (i.e. np.all(self.fixatoms_dof <= 3*self.natoms) )